Auth0/Authentication.swift

```diff
@@ -178,10 +178,12 @@ public protocol Authentication: Trackable, Loggable {
      - parameter password:          password for the new user
      - parameter connection:        name where the user will be created (Database connection)
      - parameter userMetadata:      additional userMetadata parameters that will be added to the newly created user.
+     - parameter rootAttributes:    root attributes that will be added to the newly created user.
 
      - returns: request that will yield a created database user (just email, username and email verified flag)
      */
-    func createUser(email: String, username: String?, password: String, connection: String, userMetadata: [String: Any]?) -> Request<DatabaseUser, AuthenticationError>
+    // swiftlint:disable:next function_parameter_count
+    func createUser(email: String, username: String?, password: String, connection: String, userMetadata: [String: Any]?, rootAttributes: [String: Any]?) -> Request<DatabaseUser, AuthenticationError>
```

Comment:
protocols' methods can be changed without breaking users? this is a public one.

Reply:
This looks like an optional argument, and it's placed at the end of the parameter list so I'm not sure it would be a breaking change.

Reply:
The argument is optional so in use you don't need to specify it. However, Lucho is correct any class that implements the protocol still has to support the behaviour. As the Protocol is Public then we should try to avoid breaking it.

I'll add support for this in the Public Extension so the original method will still work out of the box as the method in the Public Extension is a fallback.